core/src/main/java/bisq/core/account/sign/SignedWitnessService.java

```diff
@@ -487,8 +509,18 @@ private boolean verifyDate(SignedWitness signedWitness, long childSignedWitnessD
     ///////////////////////////////////////////////////////////////////////////////////////////
 
     @VisibleForTesting
-    void addToMap(SignedWitness signedWitness) {
-        signedWitnessMap.putIfAbsent(signedWitness.getHashAsByteArray(), signedWitness);
+    public void addToMap(SignedWitness signedWitness) {
+        P2PDataStorage.ByteArray hash = signedWitness.getHashAsByteArray();
+        signedWitnessMap.putIfAbsent(hash, signedWitness);
+
+        // We remove the entry with that hash in case we have cached it, so at the next getSignedWitnessSet
+        // call we use the updated signedWitnessMap to re-fill our cache.
+        getSignedWitnessSetCache.remove(new P2PDataStorage.ByteArray(signedWitness.getAccountAgeWitnessHash()));
+
+        // Not sure if that is needed as well, tests did succeed in both cases, but seems to be more safe to remove
+        // potential entries with hash as well. A removed item in getSignedWitnessSetCache carries no risk, though a
+        // remaining item would.
+        getSignedWitnessSetCache.remove(hash);
```

Comment:
This looks redundant, it's removing the same item from the cache again. `hash` is calculated as `new P2PDataStorage.ByteArray(signedWitness.getAccountAgeWitnessHash())` when following the calls to signedWitness.

Reply:
I have not looked closely but I took it from the getSignedWitnessSet methods where `signedWitness.getHashAsByteArray()` got compared with `new P2PDataStorage.ByteArray(signedWitness.getAccountAgeWitnessHash())`. Have not looked deeper into the domain if those 2 data are the same actually. If so, yes that second remove is pointless.

Reply:
I looked up in the code and both fields are different!
```
byte[] data = Utilities.concatenateByteArrays(accountAgeWitnessHash, signature);
data = Utilities.concatenateByteArrays(data, signerPubKey);
hash = Hash.getSha256Ripemd160hash(data);
        
public P2PDataStorage.ByteArray getHashAsByteArray() {
        return new P2PDataStorage.ByteArray(hash);
}
```